pkg/nodeoperations/controller_test.go

```diff
@@ -295,7 +295,9 @@ func Test_handleNodeMaintenance(t *testing.T) {
 
 func prepareController(objects ...runtime.Object) *Controller {
 	scheme, _ := common.PrepareScheme()
-	client := fake.NewFakeClientWithScheme(scheme, objects...)
+	client := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(objects...).WithIndex(&corev1.Pod{},
```

Comment:
I am wondering why we need to add WithIndex() here ?  I am thinking this is causing the unit-test failure of removingNodes.

Reply:
With changing NewFakeClientWithScheme to NewClientBuilder some test were failing with slogan that index "spec.nodeName" is not defined.

Reply:
is NewFakeClientWithScheme completely removed from controller-runtime?  I think the original code just want to create a fakeclient that with the provided scheme and custome resource(nodes, pod).  I am not sure the new API NewClientBuilder behaviour.